src/runtime/format.js

```diff
@@ -10,9 +10,41 @@
  * governing permissions and limitations under the License.
  */
 
+const moment = require('moment');
+const numeral = require('numeral');
+require('numeral/locales');
+
+function parseDate(format, locale, timezone) {
+  const date = moment(parseInt(format, 10));
+  if (date.isValid()) {
+    return date.utc(timezone).locale(locale);
+  }
+
+  return null;
+}
+
 module.exports = function format(fmt, args) {
-  const argArray = Array.isArray(args) ? args : [args];
-  return fmt.replace(/{(\d+)}/g, (match, number) => (typeof argArray[number] !== 'undefined'
-    ? argArray[number]
-    : match));
+  const PLACEHOLDER_REGEX = /{(\d+)}/g;
+  const NUMBER_FORMAT_REGEX = /^[0-9.,#;$E%()-]*$/gm;
+  const PLACEHOLDER_NUMBER = /#/g;
+  const argArray = Array.isArray(args.format) ? args.format : [args.format];
+  const locale = args.locale ? args.locale : '';
+  const timezone = args.timezone ? args.timezone.replace('GMT', '').replace('UTC', '') : '';
```

Comment:
- this is only needed for dates, move down.
- I would move the `timezone.replace()` to the `parseDate()` function above, because the logic of omitting the timezone for GMT/UTC is there.